### PR TITLE
New blog post about starting at Solita.

### PR DESCRIPTION
**Pull request by Juusoh**
First experiences as a new employee of Solita... and some Java Hipsterism

### CODE REVIEW

File: _posts/2015-03-02-java-and-hipster.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+I will give a few glimpses of code of the parts that I found interesting. The coding exercise was quite trivial so most of the glimpses are excerpts or modifications of existing solutions that were originally written by someone else. Our Java stack for the exercise was based on Spring and we were encouraged to try [Spring Boot](http://projects.spring.io/spring-boot/) which makes Spring web applications stand-alone by embedding Tomcat, Jetty or Undertow into the .jar file which means that getting up and running is quick and comfy and cushioned, and ponies. Of course you Node.js and Rails hipster have always had this... I have a few years of experience on Spring but I had not tried Spring Boot before which really excited me.
```
Comment (crystoll) [inline _posts/2015-03-02-java-and-hipster.md]: Rails hipsters

File: _posts/2015-03-02-java-and-hipster.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+The fronted would be written with [AngularJS](https://angularjs.org). The most productive(?) and (over?) hyped fronted framework at the moment. Some cool guys told me that they like [Facebook's REACT](http://facebook.github.io/react/) better, and because they are so hipster, they wrap it with some [ClojureScript](https://github.com/clojure/clojurescript) goodness with [Om](https://github.com/omcljs/om). I wish I was that hip too! Luckily our senior guru Arto told me not to worry. He had heard that there actually was a Hipster stack for Java developers. I googled Java and Hipster and there it was: ["JHipster - A hipster stack for Java developers. Yeoman + Maven + Spring + AngularJS in one handy generator"](https://jhipster.github.io/). That would be my stack of choice.
```
Comment (crystoll) [inline _posts/2015-03-02-java-and-hipster.md]: front end

File: _posts/2015-03-02-java-and-hipster.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+author: juuso
+excerpt: First experiences as a new employee of Solita... and some Java Hipsterism
+---
+
+I joined Solita as a software designer early this year and was soon asked to write a blog post about my experiences starting at Solita as a youngish software designer. I was lucky to start at the same day with two great tech enthusiast, one of them being a seasoned senior software designer Arto Santala, who by the way already wrote [a great wrap up of his first experiences at Solita](http://dev.solita.fi/2015/02/16/microservices.html).
+
+All the new employees of Solita get a thorough introduction to the company and as a part of the introduction, the coders get to implement a little exercise application to familiarize themselves with a typical modern web application stack. Programmers do the exercise themselves but asking for help and discussing architectural decisions with each other and with more senior employees is highly encouraged. Of the three of us who started at the same day, me and Arto were going to be very involved in Java development which meant that we would be using a Spring based stack. The third guy got a chance to delve into the world of Clojure. Lucky him?
```
Comment (pvto) [inline _posts/2015-03-02-java-and-hipster.md]: New employees at/of Solita get a thorough introduction to the company, and as a part of that coders get to

File: _posts/2015-03-02-java-and-hipster.md (hunk 1 above)
Comment (pvto) [inline _posts/2015-03-02-java-and-hipster.md]: glimpses at code

(onko trivial oikea termi, jos puhutaan useista arkkitehtuuriratkaisuista, onko tässä ristiriita?)

into a .jar file, which means

getting <your app> up and running

hipster_s_ 

before, which

File: _posts/2015-03-02-java-and-hipster.md (hunk 2 above)
Comment (pvto) [inline _posts/2015-03-02-java-and-hipster.md]: AngularJS, which many believe is the most productive front end f...

File: _posts/2015-03-02-java-and-hipster.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+The fronted would be written with [AngularJS](https://angularjs.org). The most productive(?) and (over?) hyped fronted framework at the moment. Some cool guys told me that they like [Facebook's REACT](http://facebook.github.io/react/) better, and because they are so hipster, they wrap it with some [ClojureScript](https://github.com/clojure/clojurescript) goodness with [Om](https://github.com/omcljs/om). I wish I was that hip too! Luckily our senior guru Arto told me not to worry. He had heard that there actually was a Hipster stack for Java developers. I googled Java and Hipster and there it was: ["JHipster - A hipster stack for Java developers. Yeoman + Maven + Spring + AngularJS in one handy generator"](https://jhipster.github.io/). That would be my stack of choice.
+
+## Scaffolding the app
+
+JHipster is a [Yeoman](http://yeoman.io) generator. It scaffolds beautiful modern single page AngularJS web applications running on and served by a Spring backend. Installing is done via Node Package Manager:
```
Comment (pvto) [inline _posts/2015-03-02-java-and-hipster.md]: Installation

File: _posts/2015-03-02-java-and-hipster.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+<code>
+npm install -g generator-jhipster
+</code>
+
+After the install, to get the application generated, I jus had to execute:
```
Comment (pvto) [inline _posts/2015-03-02-java-and-hipster.md]: jus_t_

File: _posts/2015-03-02-java-and-hipster.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+};
+
+utils.selectDropdownbyNum(element(by.id('departmentSelect')), 2);
+```
+
+Besides using my own utils module, I took an advantage of [a nice set of additional matchers for Jasmine](https://github.com/JamieMason/Jasmine-Matchers).
+
+## Wrapping up
+
+What I really liked about JHipster was that it provided quite complete project skeleton with most of the build automation preconfigured. Only the acceptance testing had to be set up manually. Deciding if it is a good thing to have a preconfigured and opinionated environment when you are starting a new technology is matter of everyone's own opinion. Some people prefer bottom-up learning while other prefer top-down learning.
```
Comment (pvto) [inline _posts/2015-03-02-java-and-hipster.md]: quite a complete

**Pull request merged**